### ISSUE TITLE
確率的予測機能の実装

### ISSUE THREAD

**Issue by pictureuna** [kwuking/TimeMixer#171]
## 課題
現在のTimeMixerは、時系列の点予測（point forecasting）に焦点を当てており、予測の不確実性を定量化することができません。多くの実世界のアプリケーション（金融、エネルギー需要予測、在庫管理など）では、予測値の信頼区間や確率分布を知ることが重要です。

## 提案する機能
TimeMixerに確率的予測（probabilistic forecasting）機能を追加し、予測の不確実性を定量化できるようにします。

### 1. 分布予測ヘッドの実装
```python
class ProbabilisticForecastHead(nn.Module):
    """確率的予測のためのモデルヘッド"""
    
    def __init__(self, input_dim, output_dim, distribution_type='gaussian_mixture'):
        super(ProbabilisticForecastHead, self).__init__()
        self.distribution_type = distribution_type
        
        if distribution_type == 'gaussian':
            # ガウス分布のパラメータ（平均と分散）を予測
            self.mean_projector = nn.Linear(input_dim, output_dim)
            self.std_projector = nn.Linear(input_dim, output_dim)
            
        elif distribution_type == 'gaussian_mixture':
            # ガウス混合分布のパラメータを予測
            self.n_components = 5  # 混合成分の数
            self.mean_projector = nn.Linear(input_dim, output_dim * self.n_components)
            self.std_projector = nn.Linear(input_dim, output_dim * self.n_components)
            self.weight_projector = nn.Linear(input_dim, self.n_components)
            
        elif distribution_type == 'quantile':
            # 分位点予測（例：10%, 50%, 90%）
            self.quantiles = [0.1, 0.5, 0.9]
            self.quantile_projectors = nn.ModuleList([
                nn.Linear(input_dim, output_dim) for _ in self.quantiles
            ])
    
    def forward(self, x):
        """
        確率分布パラメータの予測
        
        Args:
            x: モデルの特徴表現
            
        Returns:
            予測分布のパラメータ
        """
        if self.distribution_type == 'gaussian':
            mean = self.mean_projector(x)
            # 標準偏差は常に正の値
            log_std = self.std_projector(x)
            std = torch.exp(log_std)
            return {'mean': mean, 'std': std}
            
        elif self.distribution_type == 'gaussian_mixture':
            mean = self.mean_projector(x).view(-1, self.n_components, -1)
            log_std = self.std_projector(x).view(-1, self.n_components, -1)
            std = torch.exp(log_std)
            weights = F.softmax(self.weight_projector(x), dim=-1)
            return {'mean': mean, 'std': std, 'weights': weights}
            
        elif self.distribution_type == 'quantile':
            quantile_forecasts = [projector(x) for projector in self.quantile_projectors]
            # 各分位点の予測値を辞書にまとめる
            return {f'q{q}': forecast for q, forecast in zip(self.quantiles, quantile_forecasts)}
```

### 2. 確率的損失関数
さまざまな確率的予測手法に対応する損失関数を実装：

```python
def negative_log_likelihood_loss(y_pred, y_true):
    """ガウス分布の負の対数尤度損失"""
    mean = y_pred['mean']
    std = y_pred['std']
    
    # ガウス分布の負の対数尤度
    nll = 0.5 * torch.log(2 * math.pi * std**2) + 0.5 * ((y_true - mean) / std)**2
    return nll.mean()

def mixture_negative_log_likelihood_loss(y_pred, y_true):
    """ガウス混合分布の負の対数尤度損失"""
    means = y_pred['mean']  # [batch, components, dim]
    stds = y_pred['std']    # [batch, components, dim]
    weights = y_pred['weights']  # [batch, components]
    
    # 各成分の負の対数尤度を計算
    y_true = y_true.unsqueeze(1)  # [batch, 1, dim]
    
    # 各成分のガウス分布から尤度を計算
    exponent = -0.5 * ((y_true - means) / stds)**2
    normalizer = 1.0 / (stds * math.sqrt(2 * math.pi))
    component_likelihoods = normalizer * torch.exp(exponent)
    
    # 重み付き和を取る
    mixture_likelihood = (weights.unsqueeze(-1) * component_likelihoods).sum(dim=1)
    
    # 負の対数尤度を返す
    return -torch.log(mixture_likelihood + 1e-10).mean()

def quantile_loss(y_pred, y_true, quantiles=[0.1, 0.5, 0.9]):
    """分位点予測のためのPinball/Quantile損失"""
